homeassistant/remote.py

```diff
@@ -215,33 +215,31 @@ def __init__(self, hass, restrict_origin=None):
         self._lock = threading.Lock()
         self._unsub_listener = None
 
-    def connect(self, api):
+    def async_connect(self, api):
         """Attach to a Home Assistant instance and forward events.
 
         Will overwrite old target if one exists with same host/port.
         """
-        with self._lock:
-            if self._unsub_listener is None:
-                self._unsub_listener = self.hass.bus.listen(
-                    ha.MATCH_ALL, self._event_listener)
+        if self._unsub_listener is None:
+            self._unsub_listener = self.hass.bus.async_listen(
+                ha.MATCH_ALL, self._event_listener)
 
-            key = (api.host, api.port)
+        key = (api.host, api.port)
 
-            self._targets[key] = api
+        self._targets[key] = api
 
-    def disconnect(self, api):
+    def async_disconnect(self, api):
```

Comment:
Callbacks. Added annotation.

They are only called from the APi when setting up/deleting event forwarding.